config/routes.rb

```diff
@@ -26,6 +26,13 @@
         end
       end
       resources :room_settings, only: %i[update], param: :friendly_id
+      resources :shared_accesses, only: :destroy, path: '/shared_accesses/room', param: :friendly_id do
+        member do
+          post :create
```

Comment:
Not sure about this solution, need to find a way to send friendly_id with the POST/create request.
Could also have a hidden_field with the friendly_id in the Create form.